src/IO/RemoteFileMetaDataBase.h

```diff
@@ -0,0 +1,86 @@
+#pragma once
+#include <memory>
+#include <base/types.h>
+#include <functional>
+#include <unordered_map>
+#include <boost/core/noncopyable.hpp>
+
+namespace DB
+{
+class RemoteFileMetaDataBase
+{
+public:
+    RemoteFileMetaDataBase() = default;
+    RemoteFileMetaDataBase(const String & schema_,
+            const String & cluster_,
```

Comment:
`cluster` is a too metadata specific thing. It should not be in the base class.

Reply:
done